src/App.tsx

```diff
@@ -7,33 +8,85 @@
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type ClockProps = {
+  name: string;
+};
+
+type ClockState = {
+  time: string;
+  hasClock: boolean;
+  name: string
+};
+
+export class App extends React.Component<ClockProps, ClockState> {
+  timerId: number | null = null;
+
+  state = {
+    time: '',
```

Comment:
In the initial state, the 'name' property should be initialized with the 'name' prop passed to the component.